scripts/lib/format-converter.ts

```diff
@@ -122,3 +124,103 @@ function coerceToBoolean(input: string): boolean | string {
   if (input.toLowerCase() === 'false') return false;
   return input;
 }
+
+function colToLetter(column: number) {
+  let temp,
+    letter = '';
+  while (column > 0) {
+    temp = (column - 1) % 26;
+    letter = String.fromCharCode(temp + 65) + letter;
+    column = (column - temp - 1) / 26;
+  }
+  return letter;
+}
+
+function retrieveCellHyperLinks(cell: sheets_v4.Schema$CellData): string[] {
+  const hyperlinks = new Set<string>();
+  if (cell.textFormatRuns) {
+    for (const run of cell.textFormatRuns) {
+      const uri = run.format?.link?.uri;
+      if (uri && !cell.formattedValue!.includes(uri)) {
+        hyperlinks.add(uri);
+      }
+    }
+  }
+  if (cell.hyperlink && !cell.formattedValue!.includes(cell.hyperlink)) {
+    hyperlinks.add(cell.hyperlink);
+  }
+  return Array.from(hyperlinks);
+}
+
+export enum LinkMode {
+  Convert,
+  Drop,
+  Error,
+}
+
+/**
+ * Call this function to convert a Google Sheet to a csv-like flat format.
+ * @param incentives the Google sheet, which can be retrieved with the Sheets API.
+ * @param convertLinks if true, hyperlinks will be inserted into the cell text so they are not lost during conversion. If false, will error if there are hyperlinks.
+ * @param headerRow the 1-indexed row where the header data is (same row number as you see in Google Sheets).
+ * @returns a list of records suitable for the rest of the spreadsheet-to-json process.
+ */
+export function googleSheetToFlatData(
+  incentives: sheets_v4.Schema$Sheet,
+  convertLinks: LinkMode,
+  headerRow: number,
+): Record<string, string>[] {
+  if (!incentives.data) throw new Error('No grid data in GoogleSheet');
+
+  const cellAddressToHyperlinks: { [index: string]: string[] } = {};
+  const records: Record<string, string>[] = [];
+  const headers: string[] = [];
+  incentives.data[0].rowData!.forEach((row, rIndex) => {
+    if (rIndex < headerRow - 1) return; // skip pre-header rows
+    if (rIndex === headerRow - 1) {
+      if (!row.values)
+        throw new Error(
+          'No values found in header row, probably due to incorrect header row provided',
+        );
```

Comment:
I think we've pretty consistently avoided brace-less control flow statements; that's certainly my preference. We can add this as an eslint rule and fix it all automatically, so you don't need to fix it here; I'll follow up with the eslint change.